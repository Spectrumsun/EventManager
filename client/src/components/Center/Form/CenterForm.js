import React, { Component } from 'react';

class CenterFrom extends Component {
  state = {
    centerName: '',
    city: '',
    address: '',
    availability: '',
    facility: []
  }

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
    console.log(this.state);
  }

   onSubmit = (e) => {
     e.preventDefault();
   // this.props.onUserCreate(this.state);
   }


   render() {
     return (
       <div className="card-body">
         <form onSubmit={this.onSubmit} >
           <div className="form-row">
             <div className="form-group col-md-12">
               <h5><label htmlFor="inputEmail4">Name</label></h5>
               <input
                 type="text"
                 value={this.state.centerName}
                 onChange={this.onChange}
                 name="centerName"
                 className="form-control form-control-lg"
                 required
               />
             </div>
             <div className="form-group col-md-6">
               <h5><label htmlFor="inputPassword4">City</label></h5>
               <input
                 type="text"
                 value={this.state.city}
                 onChange={this.onChange}
                 name="city"
                 className="form-control form-control-lg"
                 required
               />
             </div>
             <div className="form-group col-md-6">
               <h5><label htmlFor="inputPassword4">Availability</label></h5>
               <input
                 type="text"
                 value={this.state.availability}
                 onChange={this.onChange}
                 name="availability"
                 className="form-control form-control-lg"
                 required
               />
             </div>
             <div className="form-group col-md-12">
               <h5><label htmlFor="inputPassword4">Address</label></h5>
               <input
                 type="text"
                 value={this.state.address}
                 onChange={this.onChange}
                 name="address"
                 className="form-control form-control-lg"
                 required
               />
             </div>
           </div>
           <h5>Add Facilities</h5>
           <input type="text" className="form-control col-md-6 form-control-lg" id="addFacility" style={{ float: 'left' }} />
           <input type="reset" value="add" id="add" className="btn btn-info btn-lg" style={{ marginLeft: '20px' }} />
           <br />
           <br />
           <ul className="list-group col-md-5" id="list" >
             <li className="list-group-item centerlist">Security</li>
           </ul>
           <br />
           <button type="submit" className="btn btn-primary btn-lg">Done</button>
           <br />
           <br />
         </form>
       </div>


     );
   }
}

export default CenterFrom;

