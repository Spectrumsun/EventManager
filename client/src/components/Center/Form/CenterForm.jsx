import React from 'react';
import uuid from 'uuid-random';
import PropTypes from 'prop-types';

/**
 * @description renders component to the DOM
 *
 * @function centerFrom
 *
 * @returns {JSX} JSX component for the form input
 */
const centerFrom = props => (
  <div className="card-body">
    <form onSubmit={props.onSubmit} onKeyPress={props.onKeyPress}>
      <div className="form-row">
        <div className="form-group col-md-12">
          <h5><label htmlFor="inputEmail4">Center Name</label></h5>
          <input
            type="text"
            value={props.name}
            onChange={props.onChange}
            name="name"
            className="form-control form-control-lg"
            required
          />
        </div>
        <div className="form-group col-md-6">
          <h5><label htmlFor="inputPassword4">City</label></h5>
          <input
            type="text"
            value={props.city}
            onChange={props.onChange}
            name="city"
            className="form-control form-control-lg"
            required
          />
        </div>
        <div className="form-group col-md-6">
          <h5>
            <label htmlFor="inputPassword4">
            Availability
            </label>
          </h5>
          <input
            type="text"
            value={props.availability}
            onChange={props.onChange}
            name="availability"
            className="form-control form-control-lg"
            required
          />
        </div>
        <div className="form-group col-md-12">
          <h5>
            <label htmlFor="inputPassword4">Address
            </label>
          </h5>
          <input
            type="text"
            value={props.address}
            onChange={props.onChange}
            name="address"
            className="form-control form-control-lg"
            required
          />
        </div>
        <div className="form-group col-md-12">
          <h5>
            <label htmlFor="inputPassword4">About
            </label>
          </h5>
          <textarea
            type="text"
            value={props.about}
            onChange={props.onChange}
            name="about"
            id="exampleFormControlTextarea1"
            rows="5"
            className="form-control form-control-lg"
            required
          />
        </div>
        <div className="form-group col-md-12">
          <h5>
            <label htmlFor="inputPassword4">Add Image</label>
          </h5>
          <input
            type="file"
            onChange={props.handleImageChange}
            name="image"
            accept="image/gif, image/png, image/jpeg"
            className="btn btn-light"
          />
        </div>
        <div className="imgPreview">
          {props.imagePreview}
        </div>
      </div>
      <h5>Add Facilities</h5>
      <input
        type="text"
        value={props.values}
        className="form-control col-md-6 form-control-lg"
        style={{ float: 'left' }}
        name="values"
        onChange={props.onChange}
      />

      <input
        type="button"
        value="add"
        id="add"
        disabled={!props.values}
        onClick={props.onClick}
        className="btn btn-info btn-lg"
        style={{ marginLeft: '20px' }}
      />

      <br />
      <br />
      <ul className="list-group col-md-6">
        {props.facility.map((list, i) =>
          (<li
            className="list-group-item d-flex justify-content-between align-items-center"
            key={uuid()}
          >
            {list}
            <span
              className="badge badge-danger badge-pill point"
              onClick={() =>
            props.removeFacility(remove)}
            >X
            </span>
           </li>))}
      </ul>
      <br />
      <button
        type="submit"
        className="btn btn-primary btn-lg"
        disabled={props.formValid}
      >Submit
      </button>
      <br />
      <br />
      <div className="progress">
        <div
          className="progress-bar progress-bar-striped"
          role="progressbar"
          style={{ width: props.progress }}
          aria-valuenow={props.progress}
          aria-valuemin="0"
          aria-valuemax="100"
        >{props.progress}
        </div>
      </div>
    </form>
  </div>
);


centerFrom.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
  progress: PropTypes.string.isRequired,
  about: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  handleImageChange: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  availability: PropTypes.string.isRequired,
  address: PropTypes.string.isRequired,
  values: PropTypes.string.isRequired,
  facility: PropTypes.array.isRequired,
  imagePreview: PropTypes.object,
  check: PropTypes.bool.isRequired
};

centerFrom.defaultProps = {
  imagePreview: {}
};

export default centerFrom;
