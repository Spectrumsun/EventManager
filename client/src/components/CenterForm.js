import React from 'react';

const centerFrom = () => (

  <div className="card-body">
    <form action="viewevent.html">
      <div className="form-row">
        <div className="form-group col-md-6">
          <h5><label htmlFor="inputEmail4">Name</label></h5>
          <input type="text" className="form-control form-control-lg" placeholder required />
        </div>
        <div className="form-group col-md-6">
          <h5><label htmlFor="inputPassword4">Address</label></h5>
          <input type="text" className="form-control form-control-lg" required />
        </div>
      </div>
      <h5>Add Facilities</h5>
      <input type="text" className="form-control col-md-6 form-control-lg" id="addFacility" style={{ float: 'left' }} />
      <input type="reset" value="add" id="add" className="btn btn-info btn-lg" style={{ marginLeft: '20px' }} />
      <br />
      <br />
      <ul className="list-group col-md-5" id="list" >
        <li className="list-group-item ">Chairs</li>
        <li className="list-group-item ">Car pack</li>
        <li className="list-group-item">Sound System</li>
      </ul>
      <br />
      <button type="submit" className="btn btn-primary btn-lg">Done</button>
      <br />
      <br />
    </form>
  </div>
);

export default centerFrom;

