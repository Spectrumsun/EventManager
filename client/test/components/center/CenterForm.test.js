import React from 'react';
import { shallow } from 'enzyme';
import CenterForm from '../../../src/components/Center/Form/CenterForm';

const props = {
  facility: ['car', 'table', 'fan'],
  onSubmit: () => { console.log('hi'); },
  onClick: () => { console.log('hi'); },
  onChange: () => { console.log('hi'); },
  progress: '100',
  about: 'olo',
  handleImageChange: () => { console.log('hi'); },
  name: 'tomato',
  city: 'lagos',
  availability: 'yes',
  address: 'no 22 yaba road',
  values: 'lolol',
  image: 'uyu',
  id: 1
};

describe('<CenterForm />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CenterForm {...props} />);
  });

  it('should render the CenterForm', () => {
    shallow(<CenterForm {...props}  />);
  });

  it('should render initial layout of CenterForm', () => {
    expect(wrapper.getElements()).toMatchSnapshot();
  });
  
  it('should have div on layout', () => {
    expect(wrapper.find('div').length).toEqual(11);
  });

  it('should have  input on layout', () => {
    expect(wrapper.find('input').length).toEqual(7);
  });


  it('should have h5 on layout', () => {
    expect(wrapper.find('h5').length).toEqual(7);
  });

  it('should have ui on layout', () => {
    expect(wrapper.find('ul').length).toEqual(1);
  });

  it('should have li on layout', () => {
    expect(wrapper.find('li').length).toEqual(3);
  });
;

  it('should have three button on layout', () => {
    expect(wrapper.find('button').length).toEqual(1);
  });
});
