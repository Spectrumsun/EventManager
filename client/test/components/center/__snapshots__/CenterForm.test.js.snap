// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CenterForm /> should render initial layout of CenterForm 1`] = `
Array [
  <div
    className="card-body"
  >
    <form
      onKeyPress={undefined}
      onSubmit={[Function]}
    >
      <div
        className="form-row"
      >
        <div
          className="form-group col-md-12"
        >
          <h5>
            <label
              htmlFor="inputEmail4"
            >
              Center Name
            </label>
          </h5>
          <input
            className="form-control form-control-lg"
            id="centerName"
            name="name"
            onChange={[Function]}
            required={true}
            type="text"
            value="tomato"
          />
        </div>
        <div
          className="form-group col-md-6"
        >
          <h5>
            <label
              htmlFor="inputPassword4"
            >
              City
            </label>
          </h5>
          <input
            className="form-control form-control-lg"
            name="city"
            onChange={[Function]}
            required={true}
            type="text"
            value="lagos"
          />
        </div>
        <div
          className="form-group col-md-6"
        >
          <h5>
            <label
              htmlFor="inputPassword4"
            >
              Availability
            </label>
          </h5>
          <input
            className="form-control form-control-lg"
            name="availability"
            onChange={[Function]}
            required={true}
            type="text"
            value="yes"
          />
        </div>
        <div
          className="form-group col-md-12"
        >
          <h5>
            <label
              htmlFor="inputPassword4"
            >
              Address
            </label>
          </h5>
          <input
            className="form-control form-control-lg"
            name="address"
            onChange={[Function]}
            required={true}
            type="text"
            value="no 22 yaba road"
          />
        </div>
        <div
          className="form-group col-md-12"
        >
          <h5>
            <label
              htmlFor="inputPassword4"
            >
              About
            </label>
          </h5>
          <textarea
            className="form-control form-control-lg"
            id="exampleFormControlTextarea1"
            name="about"
            onChange={[Function]}
            required={true}
            rows="5"
            type="text"
            value="olo"
          />
        </div>
        <div
          className="form-group col-md-12"
        >
          <h5>
            <label
              htmlFor="inputPassword4"
            >
              Add Image
            </label>
          </h5>
          <input
            accept="image/gif, image/png, image/jpeg"
            className="btn btn-light"
            id="file"
            name="image"
            onChange={[Function]}
            type="file"
          />
        </div>
        <div
          className="imgPreview"
        >
          Object {}
        </div>
      </div>
      <h5>
        Add Facilities
      </h5>
      <input
        className="form-control col-md-6 form-control-lg"
        name="values"
        onChange={[Function]}
        style={
          Object {
            "float": "left",
          }
        }
        type="text"
        value="lolol"
      />
      <input
        className="btn btn-info btn-lg"
        disabled={false}
        id="add"
        onClick={[Function]}
        style={
          Object {
            "marginLeft": "20px",
          }
        }
        type="button"
        value="add"
      />
      <br />
      <br />
      <ul
        className="list-group col-md-6"
      >
        <li
          className="list-group-item d-flex justify-content-between align-items-center"
        >
          car
          <span
            className="badge badge-danger badge-pill point"
            id="remove"
            onClick={[Function]}
          >
            X
          </span>
        </li>
        <li
          className="list-group-item d-flex justify-content-between align-items-center"
        >
          table
          <span
            className="badge badge-danger badge-pill point"
            id="remove"
            onClick={[Function]}
          >
            X
          </span>
        </li>
        <li
          className="list-group-item d-flex justify-content-between align-items-center"
        >
          fan
          <span
            className="badge badge-danger badge-pill point"
            id="remove"
            onClick={[Function]}
          >
            X
          </span>
        </li>
      </ul>
      <br />
      <div
        className="progress"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="100"
          className="progress-bar progress-bar-striped"
          role="progressbar"
          style={
            Object {
              "width": "100",
            }
          }
        >
          100
        </div>
      </div>
      <br />
      <button
        className="btn btn-primary btn-lg"
        disabled={undefined}
        id="centerSubmit"
        type="submit"
      >
        Submit
      </button>
      <br />
    </form>
  </div>,
]
`;
